networking/podenv.go

```diff
@@ -65,31 +61,21 @@ type activeNet struct {
 
 // Loads nets specified by user and default one from stage1
 func (e *podEnv) loadNets() ([]activeNet, error) {
-	nets, err := loadUserNets(e.localConfig, e.netsLoadList)
-	if err != nil {
-		return nil, err
-	}
-
 	if e.netsLoadList.None() {
-		return nets, nil
+		stderr.Printf("networking namespace with loopback only")
+		return nil, nil
 	}
 
-	if !netExists(nets, "default") && !netExists(nets, "default-restricted") {
-		var defaultNet string
-		if e.netsLoadList.Specific("default") || e.netsLoadList.All() {
-			defaultNet = DefaultNetPath
-		} else {
-			defaultNet = DefaultRestrictedNetPath
-		}
-		defPath := path.Join(common.Stage1RootfsPath(e.podRoot), defaultNet)
-		n, err := loadNet(defPath)
-		if err != nil {
-			return nil, err
-		}
-		nets = append(nets, *n)
+	nets, err := e.newNetLoader().loadNets(e.netsLoadList)
+	if err != nil {
+		return nil, err
+	}
+	netSlice := make([]activeNet, 0, len(nets))
+	for _, net := range nets {
```

Comment:
Iterating over this map may discard the original order. This is a behavioral change compared to existing code and I don't think this is really what you want (even though you probably won't notice this often at runtime). Should we just directly receive an array here without post-processing?